### PR TITLE
samples: tfm: provisioning images rewrite

### PR DESCRIPTION
**Pull request by Vge0rge**
This splits the provisioning image to two
different samples. One for the application core
and one for the network core for the NRF5340.
This is done to comform with how the other NCS
samples work. It is easier for customers to use a
separate sample as a child image instead including the child image as a Zephyr module.

Ref: NCSDK-18228

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

### CODE REVIEW

**Comment by Vge0rge** [comment]
@torsteingrindvik Please verify the sample.yml files to make sure that they make sense.

**Review by vili-nordic — APPROVED** [review]

**Review by mswarowsky — APPROVED** [review]

**Comment by Vge0rge** [comment]
I will not merge this until I get a review from @tejlmand  since he requested this change and he can give give his opinion regarding this following the same strategy as our other samples.

File: samples/tfm/provisioning_image/prj.conf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2022 Nordic Semiconductor ASA
+# Copyright (c) 2023 Nordic Semiconductor ASA
```
Comment (tejlmand) [inline samples/tfm/provisioning_image/prj.conf:2]: why is the copyright year changed ?
I assume this file is still copyrighted from 2022.

File: samples/tfm/provisioning_image_net_core/prj.conf [hunk 2]
```diff
@@ -1,7 +1,9 @@
 #
-# Copyright (c) 2022 Nordic Semiconductor ASA
+# Copyright (c) 2023 Nordic Semiconductor ASA
```
Comment (tejlmand) [inline samples/tfm/provisioning_image_net_core/prj.conf:2]: I don't think modifying an existing file should change the first year the file is copyrighted

**Review by tejlmand — COMMENTED** [review]
lgtm.

But why is the copyright year being changed ?
afaik the copyright year should still be the year the file is introduced, even if later modified.

File: samples/tfm/provisioning_image_net_core/prj.conf (hunk 2 above)
Comment (Vge0rge) [reply samples/tfm/provisioning_image_net_core/prj.conf]: I reverted all the files back :)

**Review by tejlmand — APPROVED** [review]

File: samples/tfm/provisioning_image/README.rst [hunk 3]
```diff
@@ -13,7 +13,7 @@ This sample does not include a TF-M image, it is a Zephyr image intended to be f
 After completion, the device is in the Platform Root-of-Trust (PRoT) security lifecycle state called **PRoT Provisioning**.
 For more information about the PRoT security lifecycle, see Arm's Platform Security Model 1.1 defined in the Platform Security Architecture (PSA).
 
-When built for the nrf5340dk_nrf5340_cpuapp target, this image also includes a child image for the network core (nrf5340dk_nrf5340_cpunet target).
+When built for the nrf5340dk_nrf5340_cpuapp target, this image also includes the sample :ref:`provisioning_image_net_core` as a child image for the network core (nrf5340dk_nrf5340_cpunet target).
```
Comment (mia-ko) [inline samples/tfm/provisioning_image/README.rst:16]:
```suggestion
When built for the nrf5340dk_nrf5340_cpuapp target, this image by default also includes the :ref:`provisioning image for network core<provisioning_image_net_core>` sample as a child image for the network core (nrf5340dk_nrf5340_cpunet target).
```

File: samples/tfm/provisioning_image_net_core/README.rst [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+Provisioning image for network core
+###################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+Running this provisioning image in the network core will disable debugging by enabling the APPROTECT in the UICR register.
+This sample is included by default as a child image of the :ref:`provisioning_image` sample which runs on the application core.
```
Comment (mia-ko) [inline samples/tfm/provisioning_image_net_core/README.rst:12]:
```suggestion
This sample is included by default as a child image of the :ref:`provisioning image<provisioning_image>` sample which runs on the application core.
```

File: samples/tfm/provisioning_image_net_core/README.rst [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+Provisioning image for network core
+###################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+Running this provisioning image in the network core will disable debugging by enabling the APPROTECT in the UICR register.
```
Comment (mia-ko) [inline samples/tfm/provisioning_image_net_core/README.rst:11]: Does it need more explaining when or why you would need to do this (disable debugging for network core), or is it given/clear?

**Review by mia-ko — CHANGES_REQUESTED** [review]
Just a few comments :)

File: samples/tfm/provisioning_image_net_core/README.rst (hunk 5 above)
Comment (Vge0rge) [reply samples/tfm/provisioning_image_net_core/README.rst:11]: In the readme we state that:
"The provisioning images initialize the provisioning process of a device in a manner compatible with Trusted Firmware-M (TF-M)."


The provisioning process of TF-M requires disabling the debugging access. So I hope that it is clear but let me know if you think that we can write it better.

[the remark this answers is not in the document]
Comment (mia-ko) [reply samples/tfm/provisioning_image_net_core/README.rst:11]: If you decide to add the above suggestion, you can make the following change in line 14:
Replace 'Trusted Firmware-M (TF-M)' with 'TF-M'.

**Review by mia-ko — CHANGES_REQUESTED** [review]

File: samples/tfm/provisioning_image_net_core/README.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (Vge0rge) [reply samples/tfm/provisioning_image_net_core/README.rst]: @mia-ko  Committed the suggestion and replaced the TF-M below.

File: samples/tfm/provisioning_image_net_core/README.rst [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/tfm/provisioning_image_net_core`
+
+.. include:: /includes/build_and_run.txt
+
+Testing
+=======
+The sample does not build firmware for the application core and because of that the sample cannot be tested separately.
```
Comment (mia-ko) [inline samples/tfm/provisioning_image_net_core/README.rst:34]: I missed this one in the last round, I think it's the last one from me:
```suggestion
The sample does not build firmware for the application core and thus cannot be tested separately.
```

**Review by mia-ko — APPROVED** [review]

**Review by mia-ko — COMMENTED** [review]
Btw, we're missing a changelog entry, I'm guessing this should be added there.

**Pull request merged**